bootstrap_salt/kms.py

```diff
@@ -26,9 +26,32 @@ def __init__(self, aws_profile_name, aws_region_name='eu-west-1'):
         self.conn_kms = utils.connect_to_aws(boto.kms, self)
 
     def get_key_id(self, alias):
-        aliases = self.conn_kms.list_aliases()['Aliases']
-        key_ids = [a['TargetKeyId'] for a in aliases if a['AliasName'] == "alias/{0}".format(alias)]
-        return key_ids[0] if len(key_ids) > 0 else None
+        """
+        Get a list of kms key_ids with the alias supplied
+
+        Args:
+            alias(string): The string identifier of the alias to
+                search for
+
+        Returns:
+            (string): Return the first matching key_id found, None type
+                if no key_ids were found
+        """
+        # Queries are paginated, while the results returned are truncated,
+        # and we dont have a key_id, keep getting pages
+        limit = 50
+        truncated = True
+        marker = None
+        while truncated:
+            alias_response = self.conn_kms.list_aliases(limit, marker)
+            key_ids = [a['TargetKeyId'] for a in alias_response['Aliases'] if a['AliasName'] == "alias/{0}".format(alias)]
```

Comment:
Please don't use such complex list comprehensions. This line of code does too much.
Much more readable and intent-revealing if you did something like:
```python
def match_alias(item):
    return item['AliasName'] == "alias/{0}".format(alias)

def key_id(item):
    return item['TargetKeyId']

items_with_alias = filter(match_alias, alias_response['Aliases'])
key_ids = [key_id(item) for item in items_with_alias]
```
(Although `item` needs changing to accurately describe what the values in the `alias_response` list are.)